thirdparty/hnswlib/hnswlib/hnswalg.h

```diff
@@ -335,62 +391,13 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
         // the connectivity of the graph; but we do not want to keep all of them as they won't be candidates. Thus we
         // include only a subset of filtered nodes(controlled by kAlpha) in the search path.
         float accumulative_alpha = 0.0f;
+        auto add_search_candidate = [&](Neighbor n) { return retset.insert(n, disqualified); };
```

Comment:
confused here. I see iterator use the `disqualified` to hold the candidates, but the retset seems never be used in iterator, am I right? If so, the cost of insert operation for retset can be reduced.

Reply:
for iterator, this retset is the initial search result using seed_ef and all points in `retset` are later added to `workspace->retset`. 

points in `retset`/`workspace->retset` are those ready to be returned by `iterator->Next()` and won't be searched in later iteration phase; points in `disqualified`/`workspace->to_visit` still need to be searched before they can possibly be returned by `iterator->Next()`.